src/enrollments/enrollments.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable, ConflictException } from "@nestjs/common";
+
+import { EntityManager } from "@mikro-orm/core";
+
+import { ClassroomsRepository } from "@/classrooms/classrooms.repository";
+import { StudentsRepository } from "@/students/students.repository";
+
+import { CreateEnrollmentDto, EnrollmentResponseDto } from "./enrollments.dtos";
+import { EnrollmentsRepository } from "./enrollments.repository";
+
+@Injectable()
+export class EnrollmentsService {
+  constructor(
+    private readonly em: EntityManager,
+    private readonly studentRepository: StudentsRepository,
+    private readonly classroomRepository: ClassroomsRepository,
+    private readonly enrollmentRepository: EnrollmentsRepository,
+  ) {}
+
+  async enrollStudent(createEnrollmentDto: CreateEnrollmentDto): Promise<EnrollmentResponseDto> {
+    const { studentId, classroomId } = createEnrollmentDto;
+
+    const student = await this.studentRepository.findOneOrFail({ id: studentId });
+
+    const classroom = await this.classroomRepository.findOneOrFail({ id: classroomId });
+
+    const existingEnrollment = await this.enrollmentRepository.findOne({ student, classroom });
+    if (existingEnrollment) {
+      throw new ConflictException(`Student is already enrolled in this classroom`);
+    }
+
+    const enrollment = this.enrollmentRepository.create({ student, classroom });
+
+    await this.em.persistAndFlush(enrollment);
+
+    return new EnrollmentResponseDto(enrollment.id, student.id, classroom.id);
```

Comment:
Is this doing what a serializer should be doing?

Reply:
Yes. It was shaping the response